### PR TITLE
Add a dead child if there is no ATA successor

### PR DESCRIPTION
**Pull request by morxa**
If we take an action that has no ATA successor, we know that we can no longer satisfy the specification (of undesired behavior). In this case, we have reached a good state, as the plant can do any actions without violating the specification.

Therefore, whenever we encounter an action that leads to not ATA successor, add a child with the empty canonical word, which will then correctly be labelled with TOP.

### CODE REVIEW

File: src/synchronous_product/include/synchronous_product/search.h [hunk 1]
```diff
@@ -200,16 +200,54 @@ class TreeSearch
...
-			     get_next_canonical_words(*ta_, *ata_, word, K_)) {
-				const auto word_reg = reg_a(next_word);
-				child_classes[word_reg].insert(std::move(next_word));
-				outgoing_actions[word_reg].insert(std::make_pair(region_step, symbol));
+			time_successors[word] = get_time_successors(word, K_);
+		}
+		for (const auto &symbol : ta_->get_alphabet()) {
+			std::set<std::pair<RegionIndex, CanonicalABWord<Location, ActionType>>> successors;
+			for (const auto &word : node->words) {
+				// TODO optimize repeated call of get_time_successors
```
Comment (modass) [inline src/synchronous_product/include/synchronous_product/search.h:214]: can be removed

**Review by modass — APPROVED** [review]
Nicely done!

File: src/synchronous_product/include/synchronous_product/search.h (hunk 1 above)
Comment (morxa) [reply src/synchronous_product/include/synchronous_product/search.h]: Good catch!

**Review by modass — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**